janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -498,58 +515,134 @@ private static Query numericQuery(String key, Cmp relation, Number value) {
         }
     }
 
-    private SearchParams convertQuery(Condition<?> condition, KeyInformation.StoreRetriever information) {
-        SearchParams params = new SearchParams();
+    // adapted from SolrIndex
+    private List<String> customTokenize(Analyzer analyzer, String fieldName, String value) {
+        final List<String> terms = new ArrayList<>();
+        try (CachingTokenFilter stream = new CachingTokenFilter(analyzer.tokenStream(fieldName, value))) {
+            final TermToBytesRefAttribute termAtt = stream.getAttribute(TermToBytesRefAttribute.class);
+            stream.reset();
+            while (stream.incrementToken()) {
+                terms.add(termAtt.getBytesRef().utf8ToString());
+            }
+            return terms;
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
+    }
+
+    private void tokenize(SearchParams params, final Mapping mapping, final LuceneCustomAnalyzer delegatingAnalyzer, String value, String key, JanusGraphPredicate janusgraphPredicate) {
+        final List<String> terms;
+        if (mapping == null) {
+            terms = Text.tokenize(value);
+        } else {
+            final Analyzer analyzer = delegatingAnalyzer.getWrappedAnalyzer(key);
+            if (analyzer != null) {
+                terms = customTokenize(analyzer, key, value);
+            } else {
+                switch (mapping) {
+                    case STRING:
+                        terms = Collections.singletonList(value);
+                        break;
+                    case DEFAULT:
+                    case TEXT:
+                        terms = Text.tokenize(value);
+                        break;
+                    default:
+                        throw new RuntimeException("unsupported case " + mapping);
+                }
+            }
+        }
+
+        if (terms.isEmpty()) {
+            // This might happen with very short terms
+            if (janusgraphPredicate == Text.CONTAINS_PREFIX) {
```

Comment:
Add [a test](https://codecov.io/gh/JanusGraph/janusgraph/src/customluceneanalyzers/janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java#L558...566) for this case.

Reply:
What about 
        query = new IndexQuery(store, PredicateCondition.of(ANALYZED, Text.CONTAINS_PREFIX, "To")); // very short term
        assertEquals(query.toString(), 1, tx.queryStream(query).count());

in testCustomAnalyzer() of IndexProviderTest ?

I won't include this in the next commit though, because I'm not sure it will pass travis tests for solr and elastic search (excepted if you tell me to). 
Shouldn't I just make another pr for improving test coverage ?

Reply:
mmmh, maybee this test doesn't work as I think it would (I wonder if Tom isn't tokenized into "Tam", I have to check). Also Janusgraph operators have a different meaning that what I was thinking, Got to make sure I understand them properly